### DISCUSSION TITLE
Resistors?

### DISCUSSION THREAD

**Discussion by Trilis29** [14yannick/HCPBridgeMqtt#10] [Q&A]
I see in picture 2 resistors, wiring has only 1. How it should be correctly wired? One 120ohm resistor soldered onto A+ and B-? Or 1 resistor on each line?

**Comment by apromix** [comment]
Hi Trilis29,
maybe you see in picture 2 resistors in serie, so R1+R2=120 Ohm. ;-)

**Comment by Trilis29** [comment]
Possibly yes, I have Soldered resistor between A+ and B- and conected wires, but still can‘t get connection to work to Promatic 4, so I don‘t know where is the problem ☹

**Comment by kfroeschl** [comment]
did you get an error during the busscan?
(see: https://github.com/14yannick/HCPBridgeMqtt/tree/development#installation)

or worse like me:
Initially i forgot to adjust the step-down module to only output 5V and fried my ESP32 board 😅

**Comment by apromix** [comment]
Only a hint:
It's a bit confusing, but on the RS232 side of the RS485 Converter you need connect TXD to TXD and RXD to RXD (no cross !!).  Did you connect the pins correct?

If the connection works, the TXD/RXD LEDs are still quickly flashing.

**Comment by Trilis29** [comment]
Yes I did not crossed, double checked this. Only RXD is on, but not flashing.



Maybe somehow I can check logs when ebus scan is running?

 

P.S. webpage worsk really slowly, to get /status it has to wait more than 10s, /update sometimes not load, getting error connections. ESP is not powerful enough or something is buggy?

**Comment by kfroeschl** [comment]
if you have installed the "development" branch you should could enable debugging by un-commenting the following lines:

https://github.com/14yannick/HCPBridgeMqtt/blob/bfff7a544b5def61e7cc6da2e8cb604fd7dde607/HCPBridgeESP32/platformio.ini#L32

**Comment by Trilis29** [comment]
And debug via serial or there is implemented over web?

**Reply by kfroeschl** [reply]
serial would give the most detailed information.

But you may also describe how you execute the bus-scan?

Maybe also removing the power from the Supramatic Engine and start from the beginning may help.

**Comment by Trilis29** [comment]
I have powered esp32 from auxiliary port 25V in promatic. So it is on. And turned on and of last switch, light starts blinking on motor. Ground is also common between auxiliary port, RJ cable and everything wired according diagram. I soldered resistor to pins A and B and soldered A B on top, maybe thats causing problems? I could put resistor in WAGO brick just for experiment.Sent from my iPhoneOn 2023-07-03, at 15:04, Karl Fröschl ***@***.***> wrote:﻿
serial would give the most detailed information.
But you may also describe how you execute the bus-scan?
Maybe also removing the power from the Supramatic Engine and start from the beginning may help.

**Comment by kfroeschl** [comment]
so you don't use the power-supply/GND from the RS485 BUS but from somewhere else?
https://github.com/14yannick/HCPBridgeMqtt/blob/master/Images/esp32.png

**Comment by Trilis29** [comment]
I use gnd from bus cable and 25v from another port which keeps esp always onSent from my iPhoneOn 2023-07-03, at 19:39, Karl Fröschl ***@***.***> wrote:﻿
so you don't use the power-supply/GND from the RS485 BUS but from somewhere else?
https://github.com/14yannick/HCPBridgeMqtt/blob/master/Images/esp32.png

**Comment by kfroeschl** [comment]
Did the BusScan work?

Usually you only get power on the BUS after the BusScan worked. As well the Engine won't send you any data if the BusScan did not run.

On reverse the Engine will go to error mode if a connected BUS device fails to answer/disconnect.

**Comment by Trilis29** [comment]
I will try to meter the volgate on bus pins when scanningSent from my iPhoneOn 2023-07-04, at 12:42, Karl Fröschl ***@***.***> wrote:﻿
Did the BusScan work?
Usually you only get power on the BUS after the BusScan worked. As well the Engine won't send you any data if the BusScan did not run.
On reverse the Engine will go to error mode if a connected BUS device fails to answer/disconnect.

**Comment by Trilis29** [comment]
So I got progress, connection is up. Opening the gates working, after pressing down motor IMMEDIATELY stops, bus stops working and I can't even close the gate with switch, just stuck. Any ideas?

**Comment by kfroeschl** [comment]
after power off/on the engine should work again.

Problem is the ESP32 program which fails to respond or responds with something invalid. The engine should show you an error-code now.

Which repository/branch did you install on your ESP and which ESP32 model are you using?

**Comment by Trilis29** [comment]
all solved... used developer repository for debuging, when flashed master now everything works :)
